cluster-autoscaler/processors/status/scale_up_post_processor.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+)
+
+// ScaleUpStatus is the status of a scale-up attempt. This includes information
+// on if scale-up happened, description of scale-up operation performed and
+// status of pods that took part in the scale-up evaluation.
+type ScaleUpStatus struct {
+	ScaledUp                bool
+	ScaleUpDescription      []string
+	PodsTriggeredScaleUp    []*apiv1.Pod
+	PodsRemainUnschedulable []*apiv1.Pod
+	PodsAwaitEvaluation     []*apiv1.Pod
+}
+
+// ScaleUpPostProcessor processes the state of the cluster after a scale-up.
+type ScaleUpPostProcessor interface {
+	Process(context *context.AutoscalingContext, state *ScaleUpStatus) (*context.AutoscalingContext, *ScaleUpStatus, error)
+}
+
+// NewDefaultScaleUpPostProcessor creates a default instance of ScaleUpPostProcessor.
+func NewDefaultScaleUpPostProcessor() ScaleUpPostProcessor {
+	return &ScaleUpChainPostProcessor{&EventingScaleUpPostProcessor{}}
+}
+
+// ScaleUpChainPostProcessor processes the state of the cluster after a scale-up
+// by evoking chained ScaleUpPostProcessors.
+type ScaleUpChainPostProcessor []ScaleUpPostProcessor
+
+// Process processes the state of the cluster after a scale-up by evoking chained
+// ScaleUpPostProcessors.
+func (cp *ScaleUpChainPostProcessor) Process(context *context.AutoscalingContext, state *ScaleUpStatus) (*context.AutoscalingContext, *ScaleUpStatus, error) {
+	for _, p := range *cp {
+		c, s, err := p.Process(context, state)
+		if err != nil {
+			return context, state, err
```

Comment:
Do we want to exit here? E.g. if we fail to publish an event, do we want to give up on reporting scale-up status via another channel?

Reply:
Removed